src/jttt/Core/AIStrategyFactory.java

```diff
@@ -0,0 +1,24 @@
+package jttt.Core;
+
+import java.util.HashMap;
+
+public class AIStrategyFactory {
+
+    private final HashMap<Integer, AIMoveStrategy> strategies = new HashMap<>();
+    private final int DEFAULT_STRATEGY = 0;
+    private final int BOARD_SIZE_3X3 = 9;
+    private final int BOARD_SIZE_4X4 = 16;
+
+    public AIStrategyFactory() {
+        strategies.put(DEFAULT_STRATEGY, new AlphaBetaStrategy());
+        strategies.put(BOARD_SIZE_3X3, new AlphaBetaStrategy());
+        strategies.put(BOARD_SIZE_4X4, new FourByFourAlphaBetaStrategy());
+    }
+
+    public AIMoveStrategy selectStrategyByBoardSize(int boardSize) {
+        if (strategies.get(boardSize) == null ){
```

Comment:
It might read nicer if you invert this condition, to check for not null rather than null. Then the last return statement would be the default.